server/activity/activity.model.js

```diff
@@ -166,8 +166,9 @@ class Activity extends Model {
       });
   }
 
-  remove(options = {}) {
-    if (!options.recursive) return this.destroy(options);
+  removeOrRestore(opts = {}) {
+    const { recursive, soft } = opts;
+    if (!recursive) return this.deletedAt ? this.restore(opts) : this.destroy(opts);
```

Comment:
You have this in two places, right? Maybe extract it to variable:
```js
// name is 💩 , choose something better, maybe `removeOrDestroy`
const action = (opts) => this.deletedAt ? this.restore(opts) : this.destroy(opts); 
if (!recursive) return action(opts);
```
And on line https://github.com/ExtensionEngine/tailor/pull/283/files#diff-f710cd3ec887fbb8b2e0c95f34f9ba8dR191
```js
...
 .then(() => action(opts)
...
```

Reply:
Hmm, not bad. Actually `action` name is fine with me. Suggestion? @zcuric

Reply:
Whatever, I've seen used `action` before in a similar case. Keep the consistency.

Reply:
Isn't this second occurrence actually good candidate for something like:
```js
.then(() => this.removeOrRestore({ ...opts, recursive: false }));
```
thus avoiding duplication?